tests/unit/amo/components/TestAddon.js

```diff
@@ -748,6 +787,284 @@ describe('Addon', () => {
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More extensions by Krupa');
+    });
+
+    it('displays the translator name when add-on is a dictionary', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          type: ADDON_TYPE_DICT,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More dictionaries by Krupa');
+    });
+
+    it('displays the translator name when add-on is a language pack', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          type: ADDON_TYPE_LANG,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More language packs by Krupa');
+    });
+
+    it('displays the artist name when add-on is a theme', () => {
+      const root = shallowRender({
+        addon: fakeTheme,
+        addonsByAuthors: [fakeTheme],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More themes by MaDonna');
+    });
+
+    it('displays the author name in any other cases', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          type: ADDON_TYPE_OPENSEARCH,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More add-ons by Krupa');
+    });
+
+    it('displays a plural form when extension has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More extensions by these developers');
+    });
+
+    it('displays a plural form when theme has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeTheme,
+          authors: Array(2).fill(fakeTheme.authors[0]),
+        },
+        addonsByAuthors: [fakeTheme],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More themes by these artists');
+    });
+
+    it('displays a plural form when language pack has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
+          ...fakeAddon,
+          authors: Array(2).fill(fakeAddon.authors[0]),
+          type: ADDON_TYPE_LANG,
+        },
+        addonsByAuthors: [fakeAddon],
+      });
+      expect(root.find('.AddonDescription-more-addons'))
+        .toHaveProp('header', 'More language packs by these translators');
+    });
+
+    it('displays a plural form when dictionary has multiple authors', () => {
+      const root = shallowRender({
+        addon: {
```

Comment:
I know this test suite has ancient lore from every period in history so it's not worth converting the whole thing to `shallowUntilTarget` but ideally we would. I am uneasy that you are adding a bunch of this though:

```js
const root = shallowRender({
  addon: ...,
  addonsByAuthors: ...,
});
```

I would like to see these using the `dispatch` approach along with `shallowUntilTarget`, if possible. If creating `store` each time becomes too repetitive, you can move it to `beforeEach()` and then change `_loadAddon()` to reference the global store. This is what we do in other modern tests.